ibis/sql/compiler.py

```diff
@@ -1705,7 +1705,9 @@ def _format_table(self, expr):
             is_subquery = True
 
         if is_subquery or ctx.need_aliases(expr):
-            result += ' {}'.format(ctx.get_ref(expr))
+            ref = ctx.get_ref(expr)
+            if ref is not None:
```

Comment:
If this fixes a bug, please add a test.

Reply:
There is no ref for automatically materialized join, and `get_ref` returns None.

```py
b, ap, a = batting, awards_players, alltypes

joined = (b.join(ap, ['date', 'lgID', 'yearID', 'playerID'])
            .join(a, b.date == a.date))
```

Without the condition it generates:

```sql
SELECT *
FROM (
  SELECT *
  FROM ibis_testing.`batting` t0
    ALL INNER JOIN ibis_testing.`awards_players` t1
      USING `date`, `lgID`, `yearID`, `playerID`
) None
  ALL INNER JOIN ibis_testing.`functional_alltypes` t2
    USING `date`"""
```

With the condition:

```sql
SELECT *
FROM (
  SELECT *
  FROM ibis_testing.`batting` t0
    ALL INNER JOIN ibis_testing.`awards_players` t1
      USING `date`, `lgID`, `yearID`, `playerID`
)
  ALL INNER JOIN ibis_testing.`functional_alltypes` t2
    USING `date`"""
```

I've a [TODO note](https://github.com/ibis-project/ibis/pull/1375/files#diff-0052615f1e45f91283ab6b2ac55a8e4eR303) on that, I've got lost in the context stuff.